File: JavaScript/4-partial.js
'use strict';

const partial = (fn, x) => (...args) => fn(x, ...args);

// Usage

const sum4 = (a, b, c, d) => (a + b + c + d);

const f11 = partial(sum4, 1);
const f12 = partial(f11, 2);
const f13 = partial(f12, 3);
const y1 = f13(4);
console.log(y1);

File: Python/4-partial.py

def partial(fn, x):
    return lambda *args: fn(x, *args)


def summary(a, b, c, d):
    return a + b + c + d


if __name__ == '__main__':
    f1 = partial(summary, 1)
    f2 = partial(f1, 2)
    f3 = partial(f2, 3)
    print(f3(4))
